### ISSUE TITLE
Define a method for creating a Loopback adapter for use in integration testing network resource

### ISSUE THREAD

**Issue by PlagueHO** [PowerShell/xNetworking#90]
I'm actively working on this issue now so I thought I'd set it as an issue so it can be tracked.

I'm attempting to create a cmdlet/function that will create a named Loopback adapter configured with TCP/IP.

I've made some progress with this, but at the moment it requires a file called **Devcon.exe** that is part of the WDK ([Windows Driver Kit](https://msdn.microsoft.com/en-us/windows/hardware/hh852365)) to create the Loopback adapter device. The WDK is a 800MB download (too large to download and install as part of CI) with a 2.5GB install. So I might need to locate alternate download locations for this file (it's only a 40kb).

I'll ensure everything else works and then tackle the problem of obtaining Devcon.exe.

**Comment by tysonjhayes** [comment]
Wonder if we can get devcon into a nuget or chocolatey package to just get devcon.exe? That would solve the problem of the download of the exe.

**Comment by PlagueHO** [comment]
@tysonjhayes - good idea! I took a quick scan of Chocolatey in case someone had already done it - but no dice. But I'm sure packaging this is technically easy enough though, the question comes down to whether the app is allowed to be distributed this way (licensing issue?) - I'd think so as it seems to come with lots of different drivers.

**Comment by PlagueHO** [comment]
Actually, I don't see why we shouldn't build this into a choco package- the [source code for Devcon is up on GitHub](https://github.com/Microsoft/Windows-driver-samples/tree/master/setup/devcon) anyway.

I'll see if I can package it up.

**Comment by PlagueHO** [comment]
I've created a [new repo](https://github.com/PlagueHO/devcon-choco-package) that contains a Chocolatey package definition for devcon.exe.

The repo has an AppVeyor.yml definition that clones the devcon.exe source code from the MS Repo on GitHub (linked previously), builds the DevCon.exe and then packages it into a Chocolatey package. The package then gets uploaded as an AppVeyor.yml artifact.

It isn't currently automatically pushing it to Chocolatey - I'm doing that manually right now.

I've uploaded the package to Chocolatey but it hasn't yet been reviewed so it can't be seen yet. If you have a Chocolatey account I can add you as a maintainer if you want?

**Comment by tysonjhayes** [comment]
Good find! That should make it easier if we can just pull it down from some where.

**Comment by PlagueHO** [comment]
Cheers! I'm going to also convert the Loopback adapter creation and removal code into a PS Module and upload that into PowerShell Gallery. Are these new repos I'm creating worth transferring over to you? It might be best to keep any Integration test dependencies under the same authority as the xNetworking repo I was thinking.

Anyway, Chocolatey has accepted the Devcon.portable package:
https://chocolatey.org/packages/devcon.portable/10.0.10586.0

Can install a new Loopback adapter with:
``
devcon install c:\windows\inf\netloop.inf *MSLOOP
```
I'm putting all this into the LoopbackAdapter PS Module.

**Comment by PlagueHO** [comment]
Quick update:

I had to rebuild my Chocolatey package because I needed both 32-bit and 64-bit versions of DevCon.exe in it. So I've resubmitted it - waiting for package approval now.

In the meantime I wrote a PS module for managing [Loopback Adapters](https://github.com/PlagueHO/LoopbackAdapter). I've uploaded this to PowerShell Gallery, but haven't released it yet because it requires the Chocolatey package to be available.

So once all that is done I can submit the PR with the integration tests that use the Loopback Adapter module.

**Comment by PlagueHO** [comment]
Update: Finally got my Chocolatey package accepted - they had to verify the DevCon.exe bits were actually the ones in the WDK.

So have released the [Loopback Adapter](https://www.powershellgallery.com/packages/LoopbackAdapter/1.0.0.16) module now, which is required by these tests.

**Comment by PlagueHO** [comment]
Nearly forgot about this one. Will try and get some work done on it later this week.
